pkg/collector/python/init.go

```diff
@@ -317,6 +318,9 @@ func resolvePythonExecPath(ignoreErrors bool) (string, error) {
 		log.Debugf("Executable folder is %v", _here)
 
 		embeddedPythonHome3 := filepath.Join(_here, "..", "embedded3")
+		if runtime.GOOS == "darwin" {
+			embeddedPythonHome3 = filepath.Join(_here, "../..", "embedded")
+		}
 
 		// We want to use the path-relative embedded2/3 directories above by default.
 		// They will be correct for normal installation on Windows. However, if they
```

Comment:
A thought about how this is implemented originally (not directly related to what you're adding but more to what it was already there). Not sure whether it's worth changing it at this point though.

It feels to me that we should not be relying on the existence of the `embeddedPythonHome3` folder to determine whether we need to fall back to the compile time values, but that we should rather rely on 1) was a compile-time value provided and 2) was the provided compile-time value an absolute or a relative path. Something along the lines of:

```go
if pythonHome3 == "" {
    pythonHome3 = _here
} else if !filepath.IsAbs(pythonHome3) {
    pythonHome3 = filepath.join(_here, pythonHome3)
}
```

which should be generic enough to support all OS cases.

Reply:
🤔 isn't `embeddedPythonHome3` better overall ? To me it makes more sense to do something like in term of priority
```go
if embeddedPythonHome3 {
  pythonHome3 = embeddedPythonHome3
} else if pythonHome3 == "" {
    pythonHome3 = _here
} else if !filepath.IsAbs(pythonHome3) {
    pythonHome3 = filepath.join(_here, pythonHome3)
}
```

Reply:
I feel like overall the "dynamic" value should have priority over the static one provided at compile time. Or am I missing something ?

Reply:
My point would be that if I'm setting an explicit compile time option, I'd expect that to have overriding power over anything that might be hardcoded into the program. I personally find it surprising that the option would be just used as a fallback _if some default location_ didn't work.

But this would be a breaking change so I would run this by someone else that may have more context before going with it. And if you're not convinced, we can drop this for now.